src/interface/window.c

```diff
@@ -347,6 +348,28 @@ static void window_all_wheel_input()
 			return;
 }
 
+void window_set_max_count(int value) {
+	int prev = gConfigGeneral.max_window_count;
+	int val = clamp(value, MIN_WINDOW_COUNT, MAX_WINDOW_COUNT);
+	gConfigGeneral.max_window_count = val;
+	config_save_default();
+	if (prev > val && RCT2_NEW_WINDOW >= &(g_window_list[val])) {
```

Comment:
Yeah it's supposed to, but I was having some issues getting it to work properly. It's the one that's talked about in the OP post. If I could possibly get some help with it it'd be much appreciated, or if I was allowed to create an extra variable to store the count of Windows open. It's supposed to do exactly as you described it, with the added precaution of not closing out the options window (in case the user is setting the option from it) or the important windows such as the main window or toolbars